src/DevSoc.Eventathon.Attendance/AttendanceService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Dapper;
+using DevSoc.Eventathon.Attendance;
+using DevSoc.Eventathon.Calendars.Models;
+
+namespace DevSoc.Eventathon.Data;
+
+public class AttendanceService : IAttendanceService
+{
+    private readonly IConnectionManager _connectionManager;
+
+    public AttendanceService(IConnectionManager connectionManager)
+    {
+        _connectionManager = connectionManager;
+    }
+    
+    public async void RegisterAttendance(AttendanceDefinition attendanceData)
+    {
+        using var connection = await _connectionManager.Open();
+        using var transaction = connection.BeginTransaction();
+
+        const string insertQuery = @"
+        INSERT INTO public.attendance
+        (""user_id"", ""event_id"")
+        VALUES(@UserId, @EventId);";
+
+        await connection.ExecuteAsync(insertQuery, attendanceData);
```

Comment:
we should be careful doing this, as we are reusing our HTTP layer class `AttendanceDefinition` as our data layer insertion class

Reply:
What do you think would be better? Just splitting it out so it's {attendanceData.userID, attendanceData.eventID}?

Reply:
I think I've sorted this. I've implemented it the same way we've implemented UsersRepository and UsersServices